Port hardshrink to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58661 Port silu_backward to structured
* #58665 Port gelu_backward to structured
* #58664 Port gelu to structured
* **#58663 Port hardshrink to structured**
* #58662 Port hardshrink_backward and softshrink_backward to structured



Differential Revision: [D28572531](https://our.internmc.facebook.com/intern/diff/D28572531)

Freey0 committed:
Port hardshrink to structured

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/Activation.cpp:
same deal as https://github.com/pytorch/pytorch/pull/58662/files#r636167323

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572531).

Freey0 committed:
Update on "Port hardshrink to structured"


Differential Revision: [D28572531](https://our.internmc.facebook.com/intern/diff/D28572531)

[ghstack-poisoned]

Freey0 committed:
Update on "Port hardshrink to structured"




Differential Revision: [D28572531](https://our.internmc.facebook.com/intern/diff/D28572531)

[ghstack-poisoned]

Freey0 committed:
Update on "Port hardshrink to structured"




Differential Revision: [D28572531](https://our.internmc.facebook.com/intern/diff/D28572531)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572531).

Freey0 committed:
Update on "Port hardshrink to structured"




Differential Revision: [D28572531](https://our.internmc.facebook.com/intern/diff/D28572531)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572531).
